spacy/tokens/span.pxd

```diff
@@ -7,17 +8,17 @@ from ..structs cimport SpanC
 
 cdef class Span:
     cdef readonly Doc doc
-    cdef SpanC c
+    cdef shared_ptr[SpanC] c
```

Comment:
Just leaving a note that we have to decide whether this goes in 3.3 or 4.0 and in case of the latter, whether we keep it on `develop` or a different branch.